### PR TITLE
Update poltergeist's presence with identity information.

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: resources/prosody-plugins/util.lib.lua [hunk 1]
```diff
@@ -73,8 +73,67 @@ function wrap_async_run(event,handler)
...
+-- are updating (this is the poltergeist case, where a user creates
+-- a poltergeist), optional.
+-- @param creator_group the group of the user who created the user which
+-- presence we are updating (this is the poltergeist case, where a user creates
+-- a poltergeist), optional.
+function update_presence_identity(
+    stanza, user, group, creator_user, creator_group)
+
+    -- First remove any 'identity' element if it already
+    -- exists
```
Comment (bbaldino) [inline resources/prosody-plugins/util.lib.lua]: reading this now i think it's probably worth tweaking this comment to add "so it cannot be spoofed by a client"

File: resources/prosody-plugins/mod_muc_poltergeist.lua [hunk 2]
```diff
@@ -410,8 +421,16 @@ function handle_create_poltergeist (event)
     else
         username = generate_uuid();
         store_username(room, user_id, username);
+        local context = {
+            user = {
+                id = user_id;
```
Comment (bbaldino) [inline resources/prosody-plugins/mod_muc_poltergeist.lua]: is there nothing to set for the `avatar` and `name` fields?  i think i avoided assuming these were all present, but i didn't actually think there'd be a case that didn't have them.

Comment (damencho) [reply resources/prosody-plugins/mod_muc_poltergeist.lua]: This information is already present in the same presence stanza, so no need for duplication.

Comment (bbaldino) [reply resources/prosody-plugins/mod_muc_poltergeist.lua]: right but this is trying to mimic the `jitsi_meet_context_user` object, isn't it?  which has all of those fields...so things planning on reading that will expect them to be there.  like this they'll have to check for existence and, if they're not here, potentially read them from another location?

Comment (damencho) [reply resources/prosody-plugins/mod_muc_poltergeist.lua]: Yep. Anyway, for identity, the main info is user_id and group_id, and the same data for the creator. Anything extra on top of that is not needed at the moment, but it is there, cause it is there and we copy the whole object :)

File: resources/prosody-plugins/util.lib.lua (hunk 1 above)
Comment (damencho) [reply resources/prosody-plugins/util.lib.lua]: I've added the comment.

**Pull request merged**